eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -511,16 +492,19 @@ object Helpers {
       }
       val (closingTx, closingSigned) = makeClosingTx(keyManager, commitments, localScriptPubkey, remoteScriptPubkey, remoteClosingFee)
       val signedClosingTx = Transactions.addSigs(closingTx, keyManager.fundingPublicKey(commitments.localParams.fundingKeyPath).publicKey, remoteParams.fundingPubKey, closingSigned.signature, remoteClosingSig)
-      Transactions.checkSpendable(signedClosingTx).map(x => signedClosingTx.tx).recover { case _ => throw InvalidCloseSignature(commitments.channelId, signedClosingTx.tx) }
+      Transactions.checkSpendable(signedClosingTx).map(_ => signedClosingTx.tx).recover { case _ => throw InvalidCloseSignature(commitments.channelId, signedClosingTx.tx) }
     }
 
-    def generateTx(desc: String)(attempt: Try[TransactionWithInputInfo])(implicit log: LoggingAdapter): Option[TransactionWithInputInfo] = {
-      attempt match {
-        case Success(txinfo) =>
+    /** Wraps transaction generation in a Try and filters failures to avoid one transaction negatively impacting a whole commitment. */
+    private def generateTx(desc: String)(attempt: () => Either[TxGenerationSkipped, TransactionWithInputInfo])(implicit log: LoggingAdapter): Option[TransactionWithInputInfo] = {
```

Comment:
Do we really need to make `attempt` a function?

Reply:
If we want to be able to catch exceptions it may throw, I think we do. Do you see an alternative?

Reply:
I see. No I don't.

Reply:
Actually yes:
```suggestion
    private def generateTx(desc: String)(attempt: => Either[TxGenerationSkipped, TransactionWithInputInfo])(implicit log: LoggingAdapter): Option[TransactionWithInputInfo] = {
```

Allows you to call it that way:
```scala
      val mainDelayedTx = generateTx("main-delayed-output") {
        Transactions.makeClaimDelayedOutputTx(tx, localParams.dustLimit, localRevocationPubkey, remoteParams.toSelfDelay, localDelayedPubkey, localParams.defaultFinalScriptPubKey, feeratePerKwDelayed).right.map(claimDelayed => {
          val sig = keyManager.sign(claimDelayed, keyManager.delayedPaymentPoint(channelKeyPath), localPerCommitmentPoint)
          Transactions.addSigs(claimDelayed, sig)
        })
      }
```
Instead of:
```scala
      val mainDelayedTx = generateTx("main-delayed-output")(() =>
        Transactions.makeClaimDelayedOutputTx(tx, localParams.dustLimit, localRevocationPubkey, remoteParams.toSelfDelay, localDelayedPubkey, localParams.defaultFinalScriptPubKey, feeratePerKwDelayed).right.map(claimDelayed => {
          val sig = keyManager.sign(claimDelayed, keyManager.delayedPaymentPoint(channelKeyPath), localPerCommitmentPoint)
          Transactions.addSigs(claimDelayed, sig)
        }))
```

Reply:
Done with 48db1d51a.